### ISSUE TITLE
Add possibility to set sync position independent from local player ped

### ISSUE THREAD

**Issue by EtzBetz** [altmp/altv-issues#581]
Hey! I would like to have some way of setting the sync position for a player independent from local player ped. Currently you have to set a player's ped position if you want to do some kind of custom camera stuff without the ped "in use", when you still want to sync other players etc..
This would be useful for e.g. a spectating system or some kind of advanced gamemode, which does not really rely on a single ped if it relies on one at all.

**Is your feature request related to a problem? Please describe.**
Not a problem per se, but currently when you implement a spectate or freecam system, you have to hide the ped, make it invincible and make it "static" on one position and then teleport the ped with the camera so that the sync is correct.

**Describe the solution you'd like**
I'm really not sure what the best solution would be, the first ideas i had would be to have some serverside functions like

- `Player.setSyncPosition(Vector3 position, Integer range)  // sets sync position statically on one position on the map`
- `Player.setSyncEntity(Entity entity, Integer range)  // sets sync onto a synced entity, e.g. another player or a vehicle`
- `Player.resetSyncToDefault()  // resets the player's sync to how it works by default, just like it works currently`

**Describe alternatives you've considered**
I see no alternative for this really. For spectating, an alt-function which kinda does all the spectating-work and with it the sync would be great, but that then wouldn't really solve the problem for other advanced gamemodes.

**Comment by 7Hazard** [comment]
A player ped is tied to the player. If you want to have the player in another position make a fake ped. If you want to replicate the camera of the player, move the camera based on the mouse and attach the camera to a spline https://natives.altv.mp/#/0x8609C75EC438FB3B
As of now, having the players ped somewhere else other than the ped without a camera is not possible. If you wish to be able to stream networked content somewhere other than the players position, open a separate issue about that specifically. Thank you for your contribution.

**Comment by EtzBetz** [comment]
I'm not talking about moving the ped, I'm talking about changing the streaming position for a user. The ped of the player would be untouched from this.

**Comment by Lhoerion** [comment]
Maybe making an attach to camera handle, so alt:V will track position change automatically, so stream origin will move too

**Comment by Lhoerion** [comment]
Better approach would be to bind this behaviour to rendered camera pos, so all other things are automatically taken care of.
So calling a function, where you pass camera handle, it makes sync origin bound to camera pos if that camera **is rendered**. When you reset rendered camera (to gameplay camera for example), sync point is resetted back to local player. Information about that wouldn't be lost until removed manually by proper function or when camera handle is disposed.

**Comment by EtzBetz** [comment]
Yep @Lhoerion, that's probably a better way to solve this, than what I had in mind. Voting for this^^

**Comment by 7Hazard** [comment]
We might as well go as far as to provide serverside API's to be able to give a player several networking streaming points other than the player's world position.
